crates/anvil/src/eth/backend/db.rs

```diff
@@ -71,12 +72,24 @@ where
     fn init_from_snapshot(&mut self, _snapshot: StateSnapshot) {}
 }
 
+/// Helper trait to reset the DB if it's forked
+#[auto_impl::auto_impl(Box)]
+pub trait MaybeForkedDatabase {
```

Comment:
I would appreciate if someone can think of a way to not need this trait.

I needed it because `ClientFork` calls methods on `ForkedDatabase` but when I try to pass a `ForkedDatabase` into `ClientFork` from the `setup_fork_db` the compiler complains that we're working with a `dyn Db`, so I added these methods to the `Db` trait. It seems like a chicken and egg situation but maybe someone knows how we can tell the compiler that in the relevant code paths it will always be a `ForkedDatabase`?

Reply:
I think we can just move this to the Db trait instead

Reply:
If you want, I can, but seeing how these are methods which can only be called on a `ForkedDatabase` if there's no way to completely remove it I think it's best to keep it in a separate trait. Up to you though :)

Reply:
reasonable